core/src/misc/uri.cc

```diff
@@ -47,7 +51,7 @@ URI::URI() {
 }
 
 URI::URI(const std::string& path) {
```

Comment:
@tdenniston can you please add a check here after the URI is expanded, which checks that its size is smaller than `constants::name_max_len`. In this case the URI should be set to `""`, which will be caught by the caller function. It may be a good idea also to rename `constants::name_max_len` to `constants::uri_max_len`?